llvm/test/CodeGen/RISCV/zilsd-ldst-opt-prera.mir

```diff
@@ -0,0 +1,1363 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
+# RUN: llc -mtriple=riscv32 -mattr=+zilsd -run-pass riscv-prera-zilsd-opt %s -o - | FileCheck %s
+# RUN: llc -mtriple=riscv32 -mattr=+zilsd,+zilsd-4byte-align -run-pass riscv-prera-zilsd-opt %s -o - | FileCheck %s --check-prefix=CHECK-4BYTE
+--- |
+  declare void @external_func()
+
+  define i32 @basic_load_combine(ptr %0) {
+    %2 = load i32, ptr %0, align 4
+    %3 = getelementptr inbounds i32, ptr %0, i32 1
+    %4 = load i32, ptr %3, align 4
+    %5 = add i32 %2, %4
+    ret i32 %5
+  }
+
+  define void @basic_store_combine(ptr %0, i32 %1, i32 %2) {
+    store i32 %1, ptr %0, align 4
+    %4 = getelementptr inbounds i32, ptr %0, i32 1
+    store i32 %2, ptr %4, align 4
+    ret void
+  }
+
+  define i32 @basic_load_combine_8_byte_aligned(ptr %0) {
+    %2 = load i32, ptr %0, align 8
+    %3 = getelementptr inbounds i32, ptr %0, i32 1
+    %4 = load i32, ptr %3, align 8
```

Comment:
It impossible for this load to be align 8 if the previous load 4 bytes before it is also align 8.